File: whatmakesaman/3주차/PG_92334_신고_결과_받기.py
def solution(id_list, report, k):
    answer = []
    
    #중복 신고는 1번으로 치니까 중복 제거
    new_report=list(set(report))
    
    #신고자가 신고한 사람들 목록
    a_report={}
    for id in id_list:
        a_report[id]=[]
        
    #신고당한 횟수 기록
    cnt_report={}
    for id in id_list:
        cnt_report[id]=0
        
    #신고 목록들 보면서 기록
    for r in new_report:
        reporter,reported=r.split()
        a_report[reporter].append(reported)
        cnt_report[reported]=cnt_report[reported]+1
        
    #k번 이상 신고
    k_over=[]
    for id in id_list:
        if cnt_report[id] >=k:
            k_over.append(id)
            
    # 유저별 메일 받은 횟수, 신고한 사람들 중 처리된 사람 수
    for id in id_list:
        cnt_mail=0
        for r in a_report[id]:
            if r in k_over:
                cnt_mail+=1
        answer.append(cnt_mail)
    
    
    
    return answer

File: kitttae/3주차/PG_92334_신고_결과_받기.java
import java.util.*;

class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        Set<String> set = new HashSet<>();
        
        for (String str : report) {
            set.add(str);
        }
        
        Map<String,Integer> map = new HashMap<>();
        for (String str : set) {
            String s = str.split(" ")[1];
            map.put(s, map.getOrDefault(s, 0) + 1);
        }
        
        Map<String, Integer> list = new HashMap<>();
        for (int i = 0; i < id_list.length; i++) {
            list.put(id_list[i], i);
        }
        
        int[] answer = new int[id_list.length];
        for (String str : set) {
            String[] arr = str.split(" ");
            String repoter = arr[0];
            String s = arr[1];
            
            if (map.getOrDefault(s, 0) >= k) {
                answer[list.get(repoter)]++;
            }
        }
        
        return answer;
    }
}

/*
1. 각 유저는 한 번에 한 명 유저 신고가능
    - 신고 제한X, 서로 다른 유저 계속 가능
    - 동일한 유저는 1회 처리
    
2. k번 이상 신고된 경우 이용 정지.

Map으로 키 
*/
